src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -227,18 +236,26 @@ const StudentApplication = ({ userProfile, authentication }) => {
 
   /**
    * Add an applicant to the list, identified by username
-   * @param {String} username Username for student
+   * @param {String} username Username for the new applicant
    */
   const addApplicant = async (username) => {
     try {
       // Get the profile of the selected user
-      let newApplicantProfile = await user.getProfileInfo(username);
+      const newApplicantProfile = await user.getProfileInfo(username);
       let newApplicantObject = { Profile: newApplicantProfile, OffCampusProgram: '' };
+
+      // Check if the selected user is already saved on an application in the database
+      let existingAppID = null;
+      try {
+        existingAppID = await housing.getApplicationID(username);
```

Comment:
Did we decide that it will be possible for a student to have their name on an application from a past session? If so we need to also check if the existingAppID is one from the current session or not.

Reply:
The backend part of this already handles that aspect.

The backend does all the session-code related stuff, the frontend doesn't need to send any additional information to contribute to the session-code check

Reply:
Oh, I see! Would it be more clear then to rename this service method getCurrentApplicationID or something similar?

Reply:
Good idea, will make the change now